### PR TITLE
UAF-3209 Update Asset Status Code for Retired Assets

### PR DESCRIPTION
**Pull request by mmoen**
Added new file AssetRetirementGlobal.java to override the setAssetForPersist method where the asset Status Code is set.
Added new file AssetRetirementReasonLookupableHelperServiceImpl.java to override the getReturnAnchorHtmlData method. This changes the businessObjectClassName of the href in the 'return value' link on the Asset Retirement Reason Lookup from the org/kuali to edu/arizona version of the AssetRetirementGlobal class.
Added new files AssetRetirementGlobal.xml and AssetRetirementGlobalMaintenanceDocument.xml to use the new edu/arizona version of the AssetRetirementGlobal class.
Added new file ojb-cam.xml to add a database mapping for the edu/arizona version of AssetRetirementGlobal.
Modified spring-cam.xml to point to the new files that were created.

### CODE REVIEW

File: kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/ojb-cam.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<descriptor-repository version="1.0">
```
Comment (ghost) [inline kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/ojb-cam.xml:1]: I don't see why this OJB table override is needed when the new edu.arizona BO class extends org.kuali.kfs.module.cam.businessobject.AssetRetirementGlobal and there are no changed or new database fields or relationships included in this modification.

File: kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/spring-cam.xml [hunk 2]
```diff
@@ -21,6 +21,12 @@
         <property name="dataDictionaryPackages">
             <list merge="true">
                 <value>classpath:edu/arizona/kfs/module/cam/document/datadictionary/*.xml</value>
+                <value>classpath:edu/arizona/kfs/module/cam/businessobject/datadictionary/*.xml</value>
+            </list>
+        </property>
+        <property name="databaseRepositoryFilePaths">
```
Comment (ghost) [inline kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/spring-cam.xml:27]: I don't see why this OJB table override is needed when the new edu.arizona BO class extends org.kuali.kfs.module.cam.businessobject.AssetRetirementGlobal and there are no changed or new database fields or relationships included in this modification.

**Review by ghost — COMMENTED** [review]
Initial review complete. Please see my comments.

File: kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/ojb-cam.xml (hunk 1 above)
Comment (mmoen) [reply kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/ojb-cam.xml:1]: Without the override, newBo in KualiGlobalMaintainableImpl.java will be the org/kuali version of AssetRetirementGlobal and run the wrong setAssetForPersist method.

Comment (ghost) [reply kfs-cam/src/main/resources/edu/arizona/kfs/module/cam/ojb-cam.xml:1]: We need to discuss this via Hangouts or phone as I don't understand why you would use OJB to resolve this issue. The wrong tool for the job.

**Comment by ghost** [comment]
We have agreed to use the OJB override, so approving this PR.

**Pull request merged**